<?xml version="1.0"?>
<!--
    Generative models lidar launchfile
-->
<launch>
    <arg name="model_weights_path" default="VAE_COMPRESSION_CHECKPOINT" doc="Path where the pretrained weights reside."/>
    
    <param name="param_model_weights_path" type="string" value="$(arg model_weights_path)"/>
    <rosparam param="pc_height">64</rosparam>
    <rosparam param="pc_width">128</rosparam>

    <group>
        <!-- Preprocessor node -->
        <node pkg="generative_models" name="lidar_preprocessor_node" type="preprocessor_lidar.py"></node>   
        <!-- Encoder node -->
        <node pkg="generative_models" name="encoder_node" type="encoder_lidar.py"></node>
        <!-- Decoder node -->
        <node pkg="generative_models" name="decoder_node" type="decoder_lidar.py"></node>
    </group>

</launch>
